File: 101-symmetric-tree/symmetric-tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    bool Dfs(TreeNode root1, TreeNode root2)
    {
        if (root1 is null && root2 is null) 
            return true;

        if (root1?.val != root2?.val)
            return false;

        return Dfs(root1?.left, root2?.right) && 
               Dfs(root1?.right, root2?.left);
    }

    public bool IsSymmetric(TreeNode root) {
        return Dfs(root.left, root.right);
    }
}

File: 101-symmetric-tree/symmetric-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool check(TreeNode* p, TreeNode* q)
    {
        if(!p || !q) return p == q;
        if(p->val != q->val) return false;
        bool leftC = check(p->left, q->right);
        bool rightC = check(p->right, q->left);

        return (leftC && rightC);
    }
    bool isSymmetric(TreeNode* root) 
    {
        if(!root->left && !root->right) return true;
        if(!root->left || !root->right) return false;
        return check(root->left, root->right);
    }
};